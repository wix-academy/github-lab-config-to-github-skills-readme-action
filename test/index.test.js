const process = require("process");
const cp = require("child_process");

const {
  generateReadmeFromConfig,
  isPathtoRelativeMdFile,
} = require("../generate-github-skills-readme-from-lab-config");

test("generate readme", async () => {
  const readme = await generateReadmeFromConfig(
    "test/config.yml",
    "test/course-details.md",
    "test/out/readme.md"
  );

  expect(readme).toMatchSnapshot();
});



// shows how the runner will run a javascript action with env / stdout protocol
test("test runs", () => {
  process.env["INPUT_CONFIG-FILE"] = "test/config.yml";
  process.env["INPUT_COURSE-DETAILS-FILE"] = "test/course-details.md";
  process.env["INPUT_README-FILE"] = "test/out/readme.md";
  process.env["INPUT_INLINE-MARKDOWN-LINKS"] = "false";
  const result = cp.execSync(`node index.js`, { env: process.env }).toString();
  console.log(result);
});

describe("Addon", () => {
  describe("Files & Inline links", () => {
    test("relative file logic", () => {
      expect(isPathtoRelativeMdFile('./steps/0.md')).toBe(true)
      expect(isPathtoRelativeMdFile('steps/0.md')).toBe(true)
      expect(isPathtoRelativeMdFile('https://www.google.com/steps/0.md')).toBe(false)
      expect(isPathtoRelativeMdFile('https://www.google.com')).toBe(false)
    })

    test("generate readme with inline files", async () => {
      const readme = await generateReadmeFromConfig(
        "test/config-with-files.yml",
        "test/course-details.md",
        "test/out/readme-with-files-no-inline-links.md"
      );
    
      expect(readme).toMatchSnapshot();
    });
    
    test("generate readme with inline files - inline all links options", async () => {
      const readme = await generateReadmeFromConfig(
        "test/config-with-files.yml",
        "test/course-details.md",
        "test/out/readme-with-files-inline-links.md",
        {
          inlineMDlinks: true,
        }
      );
    
      expect(readme).toMatchSnapshot();
    });

    describe("test runs", () => {
      test("with files with inline links", () => {
        process.env["INPUT_CONFIG-FILE"] = "test/config-with-files.yml";
        process.env["INPUT_COURSE-DETAILS-FILE"] = "test/course-details.md";
        process.env["INPUT_README-FILE"] = "test/out/readme-with-files-inline-links.md";
        process.env["INPUT_INLINE-MARKDOWN-LINKS"] = "true";
        const result = cp
          .execSync(`node index.js`, { env: process.env })
          .toString();
        console.log(result);
      });
  
      test("with files no inline links", () => {
        process.env["INPUT_CONFIG-FILE"] = "test/config-with-files.yml";
        process.env["INPUT_COURSE-DETAILS-FILE"] = "test/course-details.md";
        process.env["INPUT_README-FILE"] = "test/out/readme-with-files-no-inline-links.md";
        process.env["INPUT_INLINE-MARKDOWN-LINKS"] = "false";
        const result = cp
          .execSync(`node index.js`, { env: process.env })
          .toString();
        console.log(result);
      });
    })
  });
});
