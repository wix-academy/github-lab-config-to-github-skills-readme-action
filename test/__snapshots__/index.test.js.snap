// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addon Files & Inline links generate readme with inline files - inline all links options 1`] = `
"
# Learn Velo


Velo is a full-stack development platform that empowers you to rapidly build, manage and deploy professional web apps.

## Course Details

This is the description we show on the course page [here](https://github.com/wix-academy/learn-velo). This first paragraph will be featured at the top of the page so make it great.
​

​
Be sure to tell the learner about the technology you are teaching, why they should learn it, how they can use it, and what you will teach them.
​


Include information on how they can reach you for questions about the content or course. 

### Help

[Post on the #academy Slack channel](https://wix.slack.com/archives/CE2AFKXEK)

## Steps
    

<details id=0 open>
<summary><h2>01 - Welcome to Velo, Wix’s developer mode</h2></summary>

Our developer advocate Joshua Alphonse is happy to introduce you to this powerful product so you can start creating your custom solutions.


#### Watch the video

[![Welcome to Velo, Wix’s developer mode](https://img.youtube.com/vi/JyTegNu4MHw/0.jpg)](https://www.youtube.com/watch?v=JyTegNu4MHw \\"Welcome to Velo, Wix’s developer mode\\") 

## Step 0

test content to be inlined


</details>

<details id=1 >
<summary><h2>02 - A tour of Velo interface</h2></summary>

learn where to find the tools and resources you need to add custom functionality to your site using code


#### Watch the video

[![A tour of Velo interface](https://img.youtube.com/vi/rzReCvkI4gg/0.jpg)](https://www.youtube.com/watch?v=rzReCvkI4gg \\"A tour of Velo interface\\") 

## Step 0

test content to be inlined


</details>

<details id=2 >
<summary><h2>03 - 📖 Read: Velo API Introduction</h2></summary>

Use the APIs to interact with site elements, your site’s database content, Wix Apps, and external services.  The APIs also give you access to information about your site, its users, and more.



👉 [📖 Read: Velo API Introduction](https://hackernoon.com/velo-api-introduction-the-basic-things-you-should-know-about-the-velo-jc2d33uz)
</details>

<details id=3 >
<summary><h2>04 - 📺 Add Custom Interactions with JavaScript | 5:48 ⏳</h2></summary>

Learn how to use Velo to create custom interactions on your site. Visit the link to see a working example


#### Watch the video

[![📺 Add Custom Interactions with JavaScript | 5:48 ⏳](https://img.youtube.com/vi/aayDBLTKG5c/0.jpg)](https://www.youtube.com/watch?v=aayDBLTKG5c \\"📺 Add Custom Interactions with JavaScript | 5:48 ⏳\\") 


👉 [📺 Add Custom Interactions with JavaScript | 5:48 ⏳](./test/steps/01_step.md)
</details>

<details id=4 >
<summary><h2>05 - 📖 Read the getting started guide (2-3 hours 🕒)</h2></summary>

Let's take a quick look at how to get started writing your first Velo code. We'll do that by creating a traditional Hello World app - Velo style.



👉 [📖 Read the getting started guide (2-3 hours 🕒)](https://learn-code.wix.com/en/getting-started-guides/beginners-guide)
</details>

<details id=5 >
<summary><h2>06 - (Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)</h2></summary>

Take on a tutorial inside Wix editor, we'll make a collapsed paragraph expand on button click. Use a switch to change image element source. Set up a sticky header (onScroll down).



👉 [(Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)](https://learn-code.wix.com/en/step-by-step-exercises-3016544)
</details>

<details id=6 >
<summary><h2>07 - Build a to-do list</h2></summary>

The to do list will display all items in the list using a repeater. Add a text input element and button. Clicking the button or pressing the enter  key will add the item in the text input element to the to do list.



👉 [Build a to-do list](https://www.wix.com/velo/example/to-do-list)
</details>

<details id=X >
<summary><h2>08 - Start using Wix Velo today</h2></summary>

Now that you know how to use Velo enable it in one of your sites. Select a Wix Site you want to intergarte Velo into today.


#### Watch the video

[![Start using Wix Velo today](https://img.youtube.com/vi/D5D1b0J0hQE/0.jpg)](https://www.youtube.com/watch?v=D5D1b0J0hQE \\"Start using Wix Velo today\\") 


👉 [Start using Wix Velo today](https://www.wix.com/my-account/site-selector/?buttonText=Start+Now&title=Select+A+Site&actionUrl=https%3A%2F%2Feditor.wix.com%2Fhtml%2Feditor%2Fweb%2Frenderer%2Fedit%2F%7B%7BprimaryApp.idInApp%7D%7D%3FmetaSiteId%3D%7B%7BmetaSiteId%7D%7D%26editorSessionId%3D%7B%7Besi%7D%7D%26autoDevMode%3Dtrue)
</details>
"
`;

exports[`Addon Files & Inline links generate readme with inline files 1`] = `
"
# Learn Velo


Velo is a full-stack development platform that empowers you to rapidly build, manage and deploy professional web apps.

## Course Details

This is the description we show on the course page [here](https://github.com/wix-academy/learn-velo). This first paragraph will be featured at the top of the page so make it great.
​

​
Be sure to tell the learner about the technology you are teaching, why they should learn it, how they can use it, and what you will teach them.
​


Include information on how they can reach you for questions about the content or course. 

### Help

[Post on the #academy Slack channel](https://wix.slack.com/archives/CE2AFKXEK)

## Steps
    

<details id=0 open>
<summary><h2>01 - Welcome to Velo, Wix’s developer mode</h2></summary>

Our developer advocate Joshua Alphonse is happy to introduce you to this powerful product so you can start creating your custom solutions.


#### Watch the video

[![Welcome to Velo, Wix’s developer mode](https://img.youtube.com/vi/JyTegNu4MHw/0.jpg)](https://www.youtube.com/watch?v=JyTegNu4MHw \\"Welcome to Velo, Wix’s developer mode\\") 

## Step 0

test content to be inlined


</details>

<details id=1 >
<summary><h2>02 - A tour of Velo interface</h2></summary>

learn where to find the tools and resources you need to add custom functionality to your site using code


#### Watch the video

[![A tour of Velo interface](https://img.youtube.com/vi/rzReCvkI4gg/0.jpg)](https://www.youtube.com/watch?v=rzReCvkI4gg \\"A tour of Velo interface\\") 

## Step 0

test content to be inlined


</details>

<details id=2 >
<summary><h2>03 - 📖 Read: Velo API Introduction</h2></summary>

Use the APIs to interact with site elements, your site’s database content, Wix Apps, and external services.  The APIs also give you access to information about your site, its users, and more.



👉 [📖 Read: Velo API Introduction](https://hackernoon.com/velo-api-introduction-the-basic-things-you-should-know-about-the-velo-jc2d33uz)
</details>

<details id=3 >
<summary><h2>04 - 📺 Add Custom Interactions with JavaScript | 5:48 ⏳</h2></summary>

Learn how to use Velo to create custom interactions on your site. Visit the link to see a working example


#### Watch the video

[![📺 Add Custom Interactions with JavaScript | 5:48 ⏳](https://img.youtube.com/vi/aayDBLTKG5c/0.jpg)](https://www.youtube.com/watch?v=aayDBLTKG5c \\"📺 Add Custom Interactions with JavaScript | 5:48 ⏳\\") 


👉 [📺 Add Custom Interactions with JavaScript | 5:48 ⏳](./test/steps/01_step.md)
</details>

<details id=4 >
<summary><h2>05 - 📖 Read the getting started guide (2-3 hours 🕒)</h2></summary>

Let's take a quick look at how to get started writing your first Velo code. We'll do that by creating a traditional Hello World app - Velo style.



👉 [📖 Read the getting started guide (2-3 hours 🕒)](https://learn-code.wix.com/en/getting-started-guides/beginners-guide)
</details>

<details id=5 >
<summary><h2>06 - (Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)</h2></summary>

Take on a tutorial inside Wix editor, we'll make a collapsed paragraph expand on button click. Use a switch to change image element source. Set up a sticky header (onScroll down).



👉 [(Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)](https://learn-code.wix.com/en/step-by-step-exercises-3016544)
</details>

<details id=6 >
<summary><h2>07 - Build a to-do list</h2></summary>

The to do list will display all items in the list using a repeater. Add a text input element and button. Clicking the button or pressing the enter  key will add the item in the text input element to the to do list.



👉 [Build a to-do list](https://www.wix.com/velo/example/to-do-list)
</details>

<details id=X >
<summary><h2>08 - Start using Wix Velo today</h2></summary>

Now that you know how to use Velo enable it in one of your sites. Select a Wix Site you want to intergarte Velo into today.


#### Watch the video

[![Start using Wix Velo today](https://img.youtube.com/vi/D5D1b0J0hQE/0.jpg)](https://www.youtube.com/watch?v=D5D1b0J0hQE \\"Start using Wix Velo today\\") 


👉 [Start using Wix Velo today](https://www.wix.com/my-account/site-selector/?buttonText=Start+Now&title=Select+A+Site&actionUrl=https%3A%2F%2Feditor.wix.com%2Fhtml%2Feditor%2Fweb%2Frenderer%2Fedit%2F%7B%7BprimaryApp.idInApp%7D%7D%3FmetaSiteId%3D%7B%7BmetaSiteId%7D%7D%26editorSessionId%3D%7B%7Besi%7D%7D%26autoDevMode%3Dtrue)
</details>
"
`;

exports[`generate readme 1`] = `
"
# Learn Velo


Velo is a full-stack development platform that empowers you to rapidly build, manage and deploy professional web apps.

## Course Details

This is the description we show on the course page [here](https://github.com/wix-academy/learn-velo). This first paragraph will be featured at the top of the page so make it great.
​

​
Be sure to tell the learner about the technology you are teaching, why they should learn it, how they can use it, and what you will teach them.
​


Include information on how they can reach you for questions about the content or course. 

### Help

[Post on the #academy Slack channel](https://wix.slack.com/archives/CE2AFKXEK)

## Steps
    

<details id=0 open>
<summary><h2>01 - Welcome to Velo, Wix’s developer mode</h2></summary>

Our developer advocate Joshua Alphonse is happy to introduce you to this powerful product so you can start creating your custom solutions.


#### Watch the video

[![Welcome to Velo, Wix’s developer mode](https://img.youtube.com/vi/JyTegNu4MHw/0.jpg)](https://www.youtube.com/watch?v=JyTegNu4MHw \\"Welcome to Velo, Wix’s developer mode\\") 


👉 [Welcome to Velo, Wix’s developer mode](https://learn-code.wix.com/en/article/about-velo-2466982)
</details>

<details id=1 >
<summary><h2>02 - A tour of Velo interface</h2></summary>

learn where to find the tools and resources you need to add custom functionality to your site using code


#### Watch the video

[![A tour of Velo interface](https://img.youtube.com/vi/rzReCvkI4gg/0.jpg)](https://www.youtube.com/watch?v=rzReCvkI4gg \\"A tour of Velo interface\\") 


👉 [A tour of Velo interface](https://learn-code.wix.com/en/article/getting-oriented-521681)
</details>

<details id=2 >
<summary><h2>03 - 📖 Read: Velo API Introduction</h2></summary>

Use the APIs to interact with site elements, your site’s database content, Wix Apps, and external services.  The APIs also give you access to information about your site, its users, and more.



👉 [📖 Read: Velo API Introduction](https://hackernoon.com/velo-api-introduction-the-basic-things-you-should-know-about-the-velo-jc2d33uz)
</details>

<details id=3 >
<summary><h2>04 - 📺 Add Custom Interactions with JavaScript | 5:48 ⏳</h2></summary>

Learn how to use Velo to create custom interactions on your site. Visit the link to see a working example


#### Watch the video

[![📺 Add Custom Interactions with JavaScript | 5:48 ⏳](https://img.youtube.com/vi/aayDBLTKG5c/0.jpg)](https://www.youtube.com/watch?v=aayDBLTKG5c \\"📺 Add Custom Interactions with JavaScript | 5:48 ⏳\\") 


👉 [📺 Add Custom Interactions with JavaScript | 5:48 ⏳](https://www.wix.com/velo/example/hide-%26-show-elements)
</details>

<details id=4 >
<summary><h2>05 - 📖 Read the getting started guide (2-3 hours 🕒)</h2></summary>

Let's take a quick look at how to get started writing your first Velo code. We'll do that by creating a traditional Hello World app - Velo style.



👉 [📖 Read the getting started guide (2-3 hours 🕒)](https://learn-code.wix.com/en/getting-started-guides/beginners-guide)
</details>

<details id=5 >
<summary><h2>06 - (Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)</h2></summary>

Take on a tutorial inside Wix editor, we'll make a collapsed paragraph expand on button click. Use a switch to change image element source. Set up a sticky header (onScroll down).



👉 [(Optional) 👩‍💻 take on a side by side exercises (2 hours 🕑)](https://learn-code.wix.com/en/step-by-step-exercises-3016544)
</details>

<details id=6 >
<summary><h2>07 - Build a to-do list</h2></summary>

The to do list will display all items in the list using a repeater. Add a text input element and button. Clicking the button or pressing the enter  key will add the item in the text input element to the to do list.



👉 [Build a to-do list](https://www.wix.com/velo/example/to-do-list)
</details>

<details id=X >
<summary><h2>08 - Start using Wix Velo today</h2></summary>

Now that you know how to use Velo enable it in one of your sites. Select a Wix Site you want to intergarte Velo into today.


#### Watch the video

[![Start using Wix Velo today](https://img.youtube.com/vi/D5D1b0J0hQE/0.jpg)](https://www.youtube.com/watch?v=D5D1b0J0hQE \\"Start using Wix Velo today\\") 


👉 [Start using Wix Velo today](https://www.wix.com/my-account/site-selector/?buttonText=Start+Now&title=Select+A+Site&actionUrl=https%3A%2F%2Feditor.wix.com%2Fhtml%2Feditor%2Fweb%2Frenderer%2Fedit%2F%7B%7BprimaryApp.idInApp%7D%7D%3FmetaSiteId%3D%7B%7BmetaSiteId%7D%7D%26editorSessionId%3D%7B%7Besi%7D%7D%26autoDevMode%3Dtrue)
</details>
"
`;
